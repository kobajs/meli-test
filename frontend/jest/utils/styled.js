import { shallow, mount } from "enzyme";
import ThemeProvider from "../../src/theme/ThemeProvider";

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(withTheme(children), options);

  return wrapper;
}

export function withTheme(children) {
  return <ThemeProvider>{children}</ThemeProvider>;
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments);
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments);
}
