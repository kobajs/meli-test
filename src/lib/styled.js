import get from 'lodash/get'

export function theme(path) {
  return props => get(props, `theme.${path}`);
}

export function breakpoints(breakpoint) {
  return props => `@media (min-width: ${props.theme.breakpoints[breakpoint]}px)`;
}