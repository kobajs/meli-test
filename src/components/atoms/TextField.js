import styled from "styled-components";
import { theme } from "../../lib/styled";

export default styled.input`
  height: 40px;
  font-size: 18px;
  border: 0;
  outline: none;
  border-radius: 4px;
  padding: 0 ${theme('spacing.double')}px;
`;
