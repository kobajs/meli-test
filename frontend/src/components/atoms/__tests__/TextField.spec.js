import React from "react";
import renderer from "react-test-renderer";

import TextField from "../TextField";
import { withTheme } from "../../../../jest/utils/styled";

it("renders correctly", () => {
  const ThemedTextField = withTheme(<TextField />);
  const tree = renderer.create(ThemedTextField).toJSON();
  expect(tree).toMatchSnapshot();
});
