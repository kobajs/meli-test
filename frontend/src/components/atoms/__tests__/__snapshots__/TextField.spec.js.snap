// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<input
  className="TextField-ngvdtw-0 iNMyxv"
/>
`;
